girder/models/file.py

```diff
@@ -94,10 +95,19 @@ def download(self, file, offset=0, headers=True, endByte=None,
         :type extraParameters: str or None
         """
         if file.get('assetstoreId'):
-            return self.getAssetstoreAdapter(file).downloadFile(
+            fileDownload = self.getAssetstoreAdapter(file).downloadFile(
                 file, offset=offset, headers=headers, endByte=endByte,
                 contentDisposition=contentDisposition,
                 extraParameters=extraParameters)
+            if inspect.isgeneratorfunction(fileDownload):
```

Comment:
Let's make this `inspect.isgeneratorfunction(fileDownload) or inspect.isgenerator(fileDownload)`, since I believe it's valid to return either type.